src/main/java/dev/morling/onebrc/CalculateAverage_artsiomkorzun.java

```diff
@@ -413,26 +415,41 @@ public void run() {
 
                 while (chunk1.has()) {
                     long word1 = word(chunk1.position);
+                    long word2 = word(chunk1.position + 8);
+
                     long separator1 = separator(word1);
-                    long pointer1 = find(aggregates, chunk1, word1, separator1);
-                    long value1 = value(chunk1);
-                    Aggregates.update(pointer1, value1);
+                    long separator2 = separator(word2);
+
+                    long pointer = find(aggregates, chunk1, word1, word2, separator1, separator2);
+                    long value = value(chunk1);
+
+                    Aggregates.update(pointer, value);
                 }
 
                 while (chunk2.has()) {
-                    long word2 = word(chunk2.position);
+                    long word1 = word(chunk2.position);
+                    long word2 = word(chunk2.position + 8);
+
+                    long separator1 = separator(word1);
                     long separator2 = separator(word2);
-                    long pointer2 = find(aggregates, chunk2, word2, separator2);
-                    long value2 = value(chunk2);
-                    Aggregates.update(pointer2, value2);
+
+                    long pointer = find(aggregates, chunk2, word1, word2, separator1, separator2);
```

Comment:
Does this approach mean the program segfaults on hash collision?

Reply:
It checks only the first entry, if there is a collision it goes to the bottom. Find is optimistic. Put is pessimistic.